### PR TITLE
Add doc availability 

### PR DESCRIPTION
**Pull request by dakshaladia**

### CODE REVIEW

File: client/src/pages/Doctorpage.js [hunk 1]
```diff
@@ -1,12 +1,126 @@
...
+  const [doctorDetails, setDoctorDetails] = useState(null);
+  const [availabilityStartDate, setAvailabilityStartDate] = useState(new Date());
+  const [availabilityEndDate, setAvailabilityEndDate] = useState(new Date());
+
+  useEffect(() => {
+    // Fetch doctor details from the backend
+    const fetchDoctorDetails = async () => {
+      try {
+        const email = "edsnowden@mbbs.com";
+        const response = await fetch(ENDPOINTS.getDoctorByEmail(email));
```
Comment (sh1v4mr4j) [inline client/src/pages/Doctorpage.js:21]: run kar rha hai code. Email variable aaya kaha se?

File: client/src/pages/Doctorpage.js [hunk 2]
```diff
@@ -1,12 +1,126 @@
...
+
+const { Header, Content } = Layout;
+const { Title, Text } = Typography;
+
+const DoctorPage = () => {
+  const [doctorDetails, setDoctorDetails] = useState(null);
+  const [availabilityStartDate, setAvailabilityStartDate] = useState(new Date());
+  const [availabilityEndDate, setAvailabilityEndDate] = useState(new Date());
+
+  useEffect(() => {
```
Comment (sh1v4mr4j) [inline client/src/pages/Doctorpage.js]: should be in services

File: server/app/services/doctor_service.py [hunk 3]
```diff
@@ -75,3 +79,58 @@ async def set_location_for_doctor(self, doctor_email: str, address: Location):
...
+        Args:
+            email (str): The email of the doctor.
+
+        Returns:
+            dict: A dictionary containing the doctor's details or None if not found.
+        """
+        try:
+            doctor = await self.doctor_collection.find_one({"email": {"$eq": email}})
+            if doctor:
+                return {
```
Comment (sh1v4mr4j) [inline server/app/services/doctor_service.py]: see how other services are returning

File: server/app/services/doctor_service.py [hunk 4]
```diff
@@ -75,3 +79,58 @@ async def set_location_for_doctor(self, doctor_email: str, address: Location):
...
+            if existing_availability:
+                # Update the existing availability 
+                await self.availability_collection.update_one(
+                    {"doctor_email": availability.doctor_email},
+                    {"$set": availability.dict()}
+                )
+                return availability  # Return updated availability
+            else:
+                # Logic to save new availability in the database
+                await self.availability_collection.insert_one(availability.dict())
```
Comment (sh1v4mr4j) [inline server/app/services/doctor_service.py:132]: change availability to schedule.

**Comment by sh1v4mr4j** [comment]
Add screenshot of Doctor's page and please  write in Pascal case

File: client/package.json [hunk 5]
```diff
@@ -11,13 +11,14 @@
...
     "@paypal/react-paypal-js": "^8.7.0",
-    "antd": "^5.22.1",
+    "antd": "^5.22.3",
     "babel-loader": "^9.2.1",
     "bootstrap-icons": "^1.11.3",
     "cypress": "^13.16.0",
     "open-location-code": "^1.0.3",
     "prop-types": "^15.8.1",
     "react": "^18.0.0",
+    "react-calendar": "^5.1.0",
```
Comment (sh1v4mr4j) [inline client/package.json:22]: why do we need this?

File: client/src/pages/Doctorpage.js [hunk 6]
```diff
@@ -1,12 +1,126 @@
...
+      return;
+    }
+
+    // Check if the end date is before the start date
+    if (availabilityEndDate < availabilityStartDate) {
+      message.error('End date cannot be before the start date. Please select the dates again.');
+      return;
+  }
+
+    const availability = {
```
Comment (sh1v4mr4j) [inline client/src/pages/Doctorpage.js]: schedule

File: client/src/pages/Doctorpage.js [hunk 7]
```diff
@@ -1,12 +1,126 @@
...
+      doctor_email: doctorDetails.email, 
+      doctor_pincode: doctorDetails.pincode, 
+      startDate: availabilityStartDate.toISOString().split('T')[0], 
+      endDate: availabilityEndDate.toISOString().split('T')[0]   
+     
+    };
+
+    try {
+      const response = await fetch(ENDPOINTS.setDoctorAvailability, {
```
Comment (sh1v4mr4j) [inline client/src/pages/Doctorpage.js]: should be in service folder

File: client/src/pages/Doctorpage.js [hunk 8]
```diff
@@ -1,12 +1,126 @@
...
+    }
+  };
 
-function DoctorPage() {
   return (
-    <div>
-      <h1>Welcome to the Doctor Page</h1>
-      <p>This is the main page of your application.</p>
-    </div>
+    <Layout style={{ minHeight: '100vh' }}>
```
Comment (sh1v4mr4j) [inline client/src/pages/Doctorpage.js]: should be a component, in components

File: server/app/routers/doctor_router.py [hunk 9]
```diff
@@ -1,10 +1,12 @@
+from http.client import HTTPException
```
Comment (sh1v4mr4j) [inline server/app/routers/doctor_router.py]: not needed, see other's code.

File: server/app/services/doctor_service.py (hunk 4 above)
Comment (dakshaladia) [reply server/app/services/doctor_service.py:132]: its saved as doctorschedule in database. this is just the variable getting used here.

File: client/src/pages/Doctorpage.js (hunk 6 above)
Comment (dakshaladia) [reply client/src/pages/Doctorpage.js]: how does it impact? the database is saving it as doctorschedule

File: client/package.json (hunk 5 above)
Comment (dakshaladia) [reply client/package.json:22]: Doctorpage.js uses it.

File: server/app/services/doctor_service.py (hunk 3 above)
Comment (dakshaladia) [reply server/app/services/doctor_service.py]: doctor object will have many parameters. we need to return only the ones we want to show on the page.

File: client/package.json (hunk 5 above)
Comment (sh1v4mr4j) [reply client/package.json:22]: antd calendar cannot be used?

File: client/src/pages/Doctorpage.js (hunk 2 above)
Comment (sh1v4mr4j) [reply client/src/pages/Doctorpage.js]: Not removed from here

File: client/src/pages/Doctorpage.js [hunk 10]
```diff
@@ -1,12 +1,98 @@
-import React from 'react';
```
Comment (sh1v4mr4j) [inline client/src/pages/Doctorpage.js:1]: can you rename `Doctorpage.js` to `DoctorPage.jsx`. #Pascal case

File: client/screenshots/DoctorPage_setAvailability.png
Comment (sh1v4mr4j) [inline client/screenshots/DoctorPage_setAvailability.png:1]: where's the navbar?
Also, naming of the file is not correct.

**Comment by sh1v4mr4j** [comment]
Did you test this?

**Comment by sh1v4mr4j** [comment]
Did you add it to routes?

File: client/screenshots/DoctorPage_setAvailability.png
Comment (dakshaladia) [reply client/screenshots/DoctorPage_setAvailability.png:1]: navbar is present.

**Comment by dakshaladia** [comment]
> Did you add it to routes?

Its already present in the doctor_router, and main.py file

**Comment by dakshaladia** [comment]
> Did you test this?

yes, did you find any issue?

File: client/src/pages/Doctorpage.js (hunk 10 above)
Comment (dakshaladia) [reply client/src/pages/Doctorpage.js:1]: its done.

**Review by sh1v4mr4j — APPROVED** [review]

**Pull request merged**